src/main/java/homework_7/Main.java

```diff
@@ -0,0 +1,16 @@
+package homework_7;
+
+public class Main {
+    public static void main(String[] args) {
+        Cat cat = new Cat("Myrka", 12);
+        Kitten tom = new Kitten("Tom", 3, "YYY");
+
+        KittenToCatFunction kittenToCatFunction = kitten -> new Cat(kitten.getName() + "Cat", kitten.getAge() + 10);
```

Comment:
Opt: you could try type conversion in lambda (as example, favourite toy to favourite color or smth like this)